### PR TITLE
Silence excessive mkldnn logging output on tests.

### PR DESCRIPTION
**Pull request by larroy**
## Description ##

Silenced excessive logging output:

http://jenkins.mxnet-ci.amazon-ml.com/blue/rest/organizations/jenkins/pipelines/mxnet-validation/pipelines/unix-cpu/branches/PR-14940/runs/1/nodes/283/steps/749/log/?start=0

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: src/operator/nn/mkldnn/mkldnn_base.cc [hunk 1]
```diff
@@ -553,8 +553,9 @@ void OpCheck::Run(mxnet::FCompute fn, const nnvm::NodeAttrs &attrs,
   for (size_t i = 0; i < out_blobs.size(); i++)
     out_blobs[i] = outputs[i].data();
   fn(attrs, ctx, in_blobs, req, out_blobs);
-
+#ifndef NDEBUG
```
Comment (TaoLv) [inline src/operator/nn/mkldnn/mkldnn_base.cc]: Thank you for fixing this. How about changing to check the environmental variable `MXNET_MKLDNN_DEBUG`?  We used this variable in other places.

**Comment by larroy** [comment]
@pengzhao-intel I saw this excessive log output and had a 3h timeout on tests, wanted to see if the PR validation time goes down because of this, probably not related.

File: src/operator/nn/mkldnn/mkldnn_base.cc (hunk 1 above)
Comment (larroy) [reply src/operator/nn/mkldnn/mkldnn_base.cc]: @TaoLv sure, will do that.

Comment (TaoLv) [reply src/operator/nn/mkldnn/mkldnn_base.cc]: Thank you @larroy . We're thinking about adding verbose levels to `MXNET_MKLDNN_DEBUG` in the future.

**Comment by TaoLv** [comment]
@larroy Sorry just notice that you're trying to save time of CI? I remember `MXNET_MKLDNN_DEBUG` is turned on explicitly in CI. So my suggestion might not help for this case.

**Comment by pengzhao-intel** [comment]
> @larroy Sorry just notice that you're trying to save time of CI? I remember `MXNET_MKLDNN_DEBUG` is turned on explicitly in CI. So my suggestion might not help for this case.

How about turning it off because we have enough test cases in the CI for MKLDNN now?

**Comment by pengzhao-intel** [comment]
Thank you for your improvement.  Merge now.

**Pull request merged**